--- a/index.js
+++ b/index.js
@@ -37,1 +37,1 @@
-var hasBrotliSupport = 'createBrotliCompress' in zlib
+var hasZstdSupport = 'createZstdCompress' in zlib
@@ -44,4 +44,4 @@
-var SUPPORTED_ENCODING = hasBrotliSupport ? ['br', 'gzip', 'deflate', 'identity'] : ['gzip', 'deflate', 'identity']
-var PREFERRED_ENCODING = hasBrotliSupport ? ['br', 'gzip'] : ['gzip']
+var SUPPORTED_ENCODING = hasZstdSupport ? ['zstd', 'br', 'gzip', 'deflate', 'identity'] : ['br', 'gzip', 'deflate', 'identity']
+var PREFERRED_ENCODING = hasZstdSupport ? ['zstd', 'br', 'gzip'] : ['br', 'gzip']
 
-var encodingSupported = ['gzip', 'deflate', 'identity', 'br']
+var encodingSupported = ['gzip', 'deflate', 'identity', 'br', 'zstd']
@@ -61,1 +61,0 @@
-  if (hasBrotliSupport) {
@@ -69,1 +68,0 @@
-  }
@@ -218,0 +216,1 @@
+: method === 'zstd' ? zlib.createZstdCompress(opts.zstd)